src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlStoreClient.cs

```diff
@@ -243,9 +245,20 @@ internal async Task<long> MergeResourcesGetTransactionVisibilityAsync(Cancellati
                 }
                 catch (Exception e)
                 {
+                    var sqlEx = e as SqlException;
+                    if (sqlEx != null && sqlEx.Number == SqlStoreErrorCodes.MergeResourcesConcurrentCallsIsAboveOptimal)
+                    {
+                        _logger.LogWarning(e, $"Error on {nameof(MergeResourcesBeginTransactionAsync)}: MergeResources concurrent calls is above optimal. Delay={delayOnOverloadMilliseconds} milliseconds.");
+
+                        // TODO: Prepare to throw 429 instead of wait/delay when bundle code is ready
+                        await Task.Delay(delayOnOverloadMilliseconds, cancellationToken);
+                        delayOnOverloadMilliseconds = (int)(delayOnOverloadMilliseconds * (2 + (RandomNumberGenerator.GetInt32(1000) / 1000.0)));
```

Comment:
Should we set a maximum wait time and maximum number of retries before throwing an exception?

Reply:
If we set it relatively low value appropriate to API calls, then background processes start failing. If we set to value appropriate for background jobs it will never happen.